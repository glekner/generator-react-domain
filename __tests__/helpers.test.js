import {
  initializePrompts,
  getPath,
  caseNames,
  validatePrompt
} from '../src/helpers';

describe('helpers', () => {
  it('initializePrompts', () => {
    const allPrompts = initializePrompts({}, {});
    expect(allPrompts).toMatchSnapshot();

    // should return empty prompts
    const emptyPrompts = initializePrompts({
      path: 'some/path',
      name: 'name',
      redux: true
    });
    expect(emptyPrompts.length).toBe(0);

    // test validation
    const shouldFalse = validatePrompt('ss');
    const shouldTrue = validatePrompt('sss');

    expect(shouldFalse).toBeFalsy();
    expect(shouldTrue).toBeTruthy();
  });

  it('getPath', () => {
    const path = getPath('some/path', 'component name', 'index');
    expect(path).toMatchSnapshot();
  });

  it('caseNames', () => {
    const name = caseNames('coMpoNent');
    expect(name).toMatchSnapshot();
  });
});
