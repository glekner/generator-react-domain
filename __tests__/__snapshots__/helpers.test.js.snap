// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`helpers caseNames 1`] = `
Object {
  "name": "CoMpoNent",
  "name_lower": "coMpoNent",
  "name_upper": "COMPONENT",
}
`;

exports[`helpers getPath 1`] = `"some/path/ComponentName/index.js"`;

exports[`helpers initializePrompts 1`] = `
Array [
  Object {
    "default": "src/components",
    "message": "Enter your Components path",
    "name": "path",
    "type": "input",
    "validate": [Function],
  },
  Object {
    "message": "Enter your Component name",
    "name": "name",
    "type": "input",
    "validate": [Function],
  },
  Object {
    "message": "Is your Component connected to [34mRedux[39m?",
    "name": "redux",
    "type": "confirm",
  },
]
`;
